Hw12/GameImpl.cpp

```diff
@@ -0,0 +1,55 @@
+#include "GameImpl.h"
+
+std::vector<SymbolMatch> GameImpl::compare2Strings(const std::string &secretWord, const std::string &guess) const
+{
+    const size_t size = std::max(secretWord.length(), guess.length());
+
+    std::vector<SymbolMatch> fullMatch(size, SymbolMatch::NoMatch);
+    for (size_t i{}; i < secretWord.length() && i < guess.length(); ++i)
+    {
+        if (secretWord[i] == guess[i])
+            fullMatch[i] = SymbolMatch::FullMatch;
+    }
+
+    for (size_t i{}; i < secretWord.length(); ++i)
+    {
+        if (fullMatch[i] == SymbolMatch::NoMatch)
+            proceedSymbol(fullMatch, guess, secretWord[i]);
+    }
+    return fullMatch;
+}
+
+void GameImpl::proceedSymbol(std::vector<SymbolMatch> &match, const std::string &guess, const char symbol) const
+{
+    for (size_t i{}; i < guess.length(); ++i)
+    {
+        if (match[i] == SymbolMatch::NoMatch && guess[i] == symbol)
+        {
+              match[i] = SymbolMatch::OtherPosition;
+              break;
+        }            
+    }
+}
```

Comment:
It looks promising) Hopefully it does cover edge-cases like duplicating letters in the words, e.g. cocos, books.
 Secret - 'hello', guessed -'lllll', the result should be "\*\*LL\*", but not "llllll"

Reply:
it will not show "llllll", because of the break on the 29 line

Reply:
![image](https://github.com/VolodiaH/HomeWorks/assets/136325043/69c4d7f2-5de3-4a59-a192-c944d5dc083a)